CMakeLists.txt

```diff
@@ -70,14 +70,46 @@ set(CMAKE_CXX_EXTENSIONS OFF)
 message("-- C++ Standard version: ${CMAKE_CXX_STANDARD}")
 
 if ("CUDA" STREQUAL "${THRUST_DEVICE_SYSTEM}")
-  if (NOT "${CMAKE_CUDA_HOST_COMPILER}" STREQUAL "")
-    unset(CMAKE_CUDA_HOST_COMPILER CACHE)
-    message(FATAL_ERROR "Thrust tests and examples require the C++ compiler"
-        " and the CUDA host compiler to be the same; to set this compiler, please"
-        " use the CMAKE_CXX_COMPILER variable, not the CMAKE_CUDA_HOST_COMPILER"
-        " variable.")
+  if ("Feta" STREQUAL "${CMAKE_CUDA_COMPILER_ID}")
+    if (NOT "${CMAKE_CXX_COMPILER}" STREQUAL "")
+      unset(CMAKE_CXX_COMPILER CACHE)
+      message(FATAL_ERROR "You are using Feta as your CUDA C++ compiler, but have"
+          " specified a different ISO C++ compiler; Feta acts as both, so please"
+          " unset the CMAKE_CXX_COMPILER variable.")
+    endif ()
+    if (NOT "${CMAKE_CUDA_HOST_COMPILER}" STREQUAL "")
+      unset(CMAKE_CUDA_HOST_COMPILER CACHE)
+      message(FATAL_ERROR "You are using Feta as your CUDA C++ compiler, but have"
+          " specified a different host ISO C++ compiler; Feta acts as both, so"
+          " please unset the CMAKE_CUDA_HOST_COMPILER variable.")
+    endif ()
+    set(CMAKE_CXX_COMPILER ${CMAKE_CUDA_COMPILER})
+    # We don't set CMAKE_CUDA_HOST_COMPILER for Feta; if we do, CMake tries to
+    # pass `-ccbin ${CMAKE_CUDA_HOST_COMPILER}` to Feta, which it doesn't
+    # understand.
+  else () 
+    if (NOT "${CMAKE_CUDA_HOST_COMPILER}" STREQUAL "")
+      unset(CMAKE_CUDA_HOST_COMPILER CACHE)
+      message(FATAL_ERROR "Thrust tests and examples require the C++ compiler"
+          " and the CUDA host compiler to be the same; to set this compiler, please"
+          " use the CMAKE_CXX_COMPILER variable, not the CMAKE_CUDA_HOST_COMPILER"
+          " variable.")
+    endif ()
+    set(CMAKE_CUDA_HOST_COMPILER ${CMAKE_CXX_COMPILER})
+  endif ()
+
+  enable_language(CXX)
```

Comment:
This is very wrong; because we don't initialize the C++ language before forcing the host compiler, `CMAKE_CXX_COMPILER` is empty if the C++ compiler is set via an environment variable, resulting in `CMAKE_CUDA_HOST_COMPILER` ending up as the default host compiler (because it's empty) later when CUDA is enabled.

Reply:
Fixed, see update.